src/components/Sidebar.tsx

```diff
@@ -1,56 +1,86 @@
+"use client";
+
 import Link from "next/link";
-import Image from "next/legacy/image";
+import React, { useContext, useState } from "react";
 import { signOut } from "next-auth/react";
+import {
+  faArrowRightFromBracket,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { UserContext } from "src/context/UserProvider";
+
+interface Button {
+  name: string;
+  icon: IconDefinition;
+  link: string;
+}
 
-const SidebarItem = ({ label }: { label: string }) => {
+export interface ISideBar {
+  buttons: Button[];
+}
+
+const SidebarItem = ({
+  label,
+  iconName,
+}: {
+  label: string;
+  iconName: IconDefinition;
+}) => {
   return (
-    <div className="text-md w-full p-2 px-4 text-left font-serif duration-150 hover:bg-darker-tan">
-      <div className="duration-150 hover:translate-x-1">{label}</div>
+    <div className="flex w-full space-x-4 text-left font-serif duration-150 hover:translate-x-1">
+      <div className="flex w-1/6 items-center justify-center">
+        <FontAwesomeIcon icon={iconName} size="lg" />
+      </div>
+      <div className="flex w-5/6 items-center text-xl">{label}</div>
```

Comment:
I changed to `text-xl` here instead of using the size in the Figma, otherwise it looks too small.